### PR TITLE
New Component - Chip

### LINKED ISSUE #1685: New Component - Chip

### PR DESCRIPTION
**Pull request by petertgiles**
This branch adds a new chip component
- Sends event up when dismissed
- Adds storybook stories
- Based on the pill component
- Only dismissed when the X is clicked, not the entire pill
- Similar to provided wireframe
image

The wireframe has chips with no colour in them so I created a new "neutral" set of colours in the `colorMap` objects.  They're OK but it seems like the gray colours in our hydrogen config file are not matching the Design Foundations document so it's not exactly the same.

Closes #1685

### CODE REVIEW

**Comment by tristan-orourke** [comment]
@JoshBeveridge can you comment on the colour? Also, lets try out the chromatic UI Review! I requested your review, you should have gotten an email? Or try following the UI Review link in the github checks list. Or just see the hosted storybook [here](https://61e099a1bb1465003a73d3ce-ghhrsrqdtz.chromatic.com/?path=/story/components-chip--chip-dismiss&args=color:primary)

File: common/src/components/Chip/Chip.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { XCircleIcon } from "@heroicons/react/outline";
+import Pill from "../Pill";
+
+export interface ChipProps extends React.HTMLProps<HTMLElement> {
+  /** The style type of the element. */
+  color: "primary" | "secondary" | "neutral";
+  /** The style mode of the element. */
+  mode: "solid" | "outline";
+  /** Handler for clicking the dismiss button in the chip */
+  onDismiss?: React.MouseEventHandler<Element>;
```
Comment (petertgiles) [inline common/src/components/Chip/Chip.tsx:11]: I looking at @tristan-orourke 's comment [here](https://github.com/GCTC-NTGC/gc-digital-talent/pull/1780#discussion_r787160287) and I'm thinking about this event and what should actually be sent up.  Just the label?  Should there be an `id` prop to have something unique to send back?

**Comment by JoshBeveridge** [comment]
Yeah, we shouldn't be deviating from the config unless there's a specific reason. H2 should provide access to `[light]gray` if you've already set a gray value, so perhaps try that.

**Comment by JoshBeveridge** [comment]
Unrelated, but for the padding, you might want to try something like `data-h2-padding="b(top-bottom, xxs) b(right-left, s)"`

**Comment by petertgiles** [comment]
> we shouldn't be deviating from the config unless there's a specific reason

I'm a bit stuck on this.  In the wireframe the background is #F0F0F0 and the foreground is #1E1E23 but neither of those colours are in the palette in the [design foundations document](https://www.figma.com/file/5w15OeN9YiOR4UX0XKW7so/GC-Talent---Design-documentation?node-id=201%3A1384) but they are in our current hydrogen config files.  Am I looking at the wrong foundations document?

File: common/src/components/Chip/Chip.tsx (hunk 1 above)
Comment (tristan-orourke) [reply common/src/components/Chip/Chip.tsx:11]: Actually on a very generic component like this, I'd keep it the way you have it.

Comment (tristan-orourke) [reply common/src/components/Chip/Chip.tsx:11]: The parent of this chip component can embed any data it needs in the dismiss handler function itself.
```
someList.map(
  ({id}) => {
    const handler = () => doSomethingWithId(id); // id is now part of the function closure
    return <Chip onDismiss={handler} />;
  }
);
```

**Review by yonikid15 — COMMENTED** [review]
Looks great Peter 👍 I can approve once that colour palette question gets resolved

**Comment by petertgiles** [comment]
> that colour palette question

Just had a quick call with Josh about this.  He's going to get with Brooke and they're going to update us with the official colours to use.  I'll have a better idea of the way forward after that.

**Comment by yonikid15** [comment]
> > that colour palette question
> 
> Just had a quick call with Josh about this. He's going to get with Brooke and they're going to update us with the official colours to use. I'll have a better idea of the way forward after that.

In that case, I'll just approve and merge this now. We can create a new issue to fix all the colour configuration once we figure that out.

**Review by yonikid15 — APPROVED** [review]

**Pull request merged**